main/inc/lib/document.lib.php

```diff
@@ -6694,4 +6711,113 @@ public static function resizeImageSlideShow(
 
         return $result;
     }
+
+    /**
+     * Adds a cloud link to the database
+     *
+     * @author - Aquilino Blanco Cores <aqblanco@gmail.com>
+     * @param array $_course
+     * @param string $path
+     * @param string $url
+     * @return int id of document or 0 if already exists or there was a problem creating it
+     */
+    public static function addCloudLink($_course, $path, $url, $name)
+    {
+        require_once api_get_path(LIBRARY_PATH) . 'fileUpload.lib.php';
+        $file_path = $path;
+        if (!self::cloudLinkExists($_course, $path, $url)) {
+            $doc_id = add_document($_course, $file_path, "link", 0, $name, $url);
+            if ($doc_id) {
+                // Update document item_property
+                api_item_property_update($_course, TOOL_DOCUMENT, $doc_id, 'DocumentAdded', api_get_user_id(), api_get_group_id(), api_get_user_id(), null, null, api_get_session_id());
+            }
+    
+            // If the file is in a folder, we need to update all parent folders
+            item_property_update_on_folder($_course, $file_path, api_get_user_id());
+    
+            return $doc_id;
+        } else {
+            return 0;
+        }
+    }
+ 
+    /**
+     * Deletes a cloud link from the database
+     *
+     * @author - Aquilino Blanco Cores <aqblanco@gmail.com>
+     * @param array $_course
+     * @param string $id
+     * @param string $url
+     * @return boolean true if success / false if an error occurred
+     */
+    public static function deleteCloudLink($_course, $id)
+    {
+ 
+        if (empty($id)) {
+            return false;
+        }
+ 
+        $document_id = $id;
+ 
+        $file_deleted_from_db = false;
+ 
+        if ($document_id) {
+            self::deleteDocumentFromDb($document_id, array(),  0, true);
+            //checking
+            $TABLE_DOCUMENT = Database::get_course_table(TABLE_DOCUMENT);
+            $sql = "SELECT * FROM $TABLE_DOCUMENT WHERE id = $document_id";
+            $result = Database::query($sql);
+            $exists = Database::num_rows($result) > 0;
+            $file_deleted_from_db = !$exists ;
+        }
+
+        return $file_deleted_from_db;
+    }
+ 
+    /**
+     * Gets the id of a cloud link with a given path
+     *
+     * @author - Aquilino Blanco Cores <aqblanco@gmail.com>
+     * @param array $_course
+     * @param string $path
+     * @param string $url
+     * @return int id of link / false if no link found
+     * @deprecated
+     */
+    public static function getCloudLinkId($_course, $path, $url)
+    {
+        $TABLE_DOCUMENT = Database::get_course_table(TABLE_DOCUMENT);
+        $course_id = $_course['real_id'];
```

Comment:
Same here. Use intval(), even if you know where $_course comes from, the function might later be called from an untrusted code.